### PR TITLE
add license reporting process

### PR DESCRIPTION
**Pull request by j1m-ryan**
### Proposed changes

Adds the license reporting writing process

### Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [ ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: internal/license_reporting/license_reporting.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
+package licensereporting
```
Comment (vepatel) [inline internal/license_reporting/license_reporting.go:1]: @j1m-ryan is it possible to have some sort of unit tests for this one?

Comment (j1m-ryan) [reply internal/license_reporting/license_reporting.go:1]: Yeah I will add some

**Review by shaun-nx — APPROVED** [review]

**Review by jjngx — APPROVED** [review]

File: internal/configs/version1/nginx-plus.tmpl [hunk 2]
```diff
@@ -358,8 +358,15 @@ stream {
     include /etc/nginx/stream-conf.d/*.conf;
 }
 
-{{- if (.NginxVersion.PlusGreaterThanOrEqualTo "nginx-plus-r31") }}
+{{- if .NginxVersion.PlusGreaterThanOrEqualTo "nginx-plus-r33" }}
+mgmt {
+    usage_report endpoint=product.connect.nginx.com interval=24h;
+    uuid_file /var/lib/nginx/nginx.id;
+    license_token conf/license.jwt;
+    deployment_ctx /etc/nginx/reporting/tracking.info;
```
Comment (shaun-nx) [inline internal/configs/version1/nginx-plus.tmpl:366]: It looks like this was changed to `deployment_context`

```
[emerg] 39#39: unknown directive "deployment_ctx" in /etc/nginx/nginx.conf:155
```

File: internal/configs/version1/nginx-plus.tmpl [hunk 3]
```diff
@@ -358,8 +358,15 @@ stream {
...
+}
+{{- else if .NginxVersion.PlusGreaterThanOrEqualTo "nginx-plus-r31" }}
 mgmt {
     usage_report interval=0s;
 }
-{{- end}}
+{{- end }}
```
Comment (shaun-nx) [inline internal/configs/version1/nginx-plus.tmpl:372]: @j1m-ryan do you think it still makes sense to keep this default `mgmt` block here?
From what I remember, we added this to confirm that the existence of the `mgmt` block wouldn't cause any issues.

Comment (j1m-ryan) [reply internal/configs/version1/nginx-plus.tmpl:372]: No it doesn't, this was as a flag for us, if we merged before adding R33 to the build

Comment (AlexFenlon) [reply internal/configs/version1/nginx-plus.tmpl:372]: We added this for users of older NGINX Plus versions to build the image, the next release will remove this and only have the r33 related block there.

**Review by pdabelf5 — APPROVED** [review]

**Review by jjngx — APPROVED** [review]

**Pull request merged**